src/main/scala/piconot/internal/Empty.scala

```diff
@@ -0,0 +1,46 @@
+package piconot.internal
+
+import picolib.semantics._
+
+import scala.language.postfixOps
+
+/**
+  * Created by tmf on 10/9/16.
+  */
+object EmptyBot extends PicoWhile("resources" / "empty.txt") {
+
```

Comment:
Even though you had to make compromises on your internal DSL, I think it's definitely more user-friendly than the sample language we were given. From talking with Prof. Ben, it seemed like pattern matching may have been the most feasible thing to implement your ideal language as an internal DSL, but that would be really tedious!